marlowe-runtime/src/Language/Marlowe/Runtime/Transaction/Api.hs

```diff
@@ -0,0 +1,363 @@
+{-# LANGUAGE EmptyDataDeriving #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Language.Marlowe.Runtime.Transaction.Api
+  where
+
+import Cardano.Api
+  ( AsType(..)
+  , ScriptDataSupportedInEra(..)
+  , SerialiseAsRawBytes(serialiseToRawBytes)
+  , Tx
+  , TxBody
+  , deserialiseFromCBOR
+  , deserialiseFromRawBytes
+  , serialiseToCBOR
+  )
+import Cardano.Api.Shelley (StakeCredential(..))
+import qualified Data.Aeson as Aeson
+import Data.Binary (Binary, Get, get, getWord8, put)
+import Data.Binary.Put (Put, putWord8)
+import Data.ByteString (ByteString)
+import Data.Map (Map)
+import Data.Set (Set)
+import Data.Some (Some(..))
+import Data.Time (UTCTime)
+import Data.Type.Equality (type (:~:)(Refl))
+import Data.Void (Void, absurd)
+import GHC.Generics (Generic)
+import Language.Marlowe.Runtime.ChainSync.Api (Address, BlockHeader, TokenName, TxId, getUTCTime, putUTCTime)
+import Language.Marlowe.Runtime.Core.Api
+import Language.Marlowe.Runtime.Transaction.Constraints (UnsolvableConstraintsError)
+import Network.Protocol.Job.Types (Command(..), SomeTag(..))
+
+-- | The low-level runtime API for building and submitting transactions.
+data MarloweTxCommand status err result where
```

Comment:
Is signing part of this design? Would it be appropriate for a `Sign` constructor to trigger a handoff to the actual signing? (Probably not, but I thought I'd mention this possiblity.)

Reply:
Signing is not part of this design. The runtime returns an unsigned TxBody for you to sign however you want, then accepts a signed Tx body in return. This allows wallet integration to be fully handled by clients of the runtime.

The CLI on the other hand should provide integrations with different wallet types (this might be our CIP-30 client actually).